corporate/views.py

```diff
@@ -159,12 +179,23 @@ def initial_upgrade(request: HttpRequest) -> HttpResponse:
         return render(request, "404.html", status=404)
 
     billing_page_url = reverse(billing_home)
+    if request.GET.get("onboarding") is not None:
+        billing_page_url = f"{billing_page_url}?onboarding=true"
+
+    realm_on_free_trial = False
 
     customer = get_customer_by_realm(user.realm)
-    if customer is not None and (get_current_plan_by_customer(customer) is not None or customer.sponsorship_pending):
-        if request.GET.get("onboarding") is not None:
-            billing_page_url = f"{billing_page_url}?onboarding=true"
-        return HttpResponseRedirect(billing_page_url)
+    if customer:
+        plan = get_current_plan_by_customer(customer)
+        if plan:
+            if plan.is_free_trial():
+                if not plan.charge_automatically or customer_has_credit_card_as_default_source(customer):
+                    return HttpResponseRedirect(billing_page_url)
```

Comment:
Why do we redirect to the billing page in this case?

Reply:
If their plan is not to charge_automatically. It means either
* They selected bill by invoice when they signed up for free trial using the old flow.
* Or they choose bill by invoice after they signed up for free trial in the new version

If customer has credit card on file
* It means they added a credit card when they signed up for free trial in old flow.
* They added a credit after they signed up for free trial in new flow

So in both the cases, we redirect them to billing since they already have a payment method on file. Does that makes sense?

Reply:
Makes sense, thanks for clarifiying